### PR TITLE
MM-29734 Add end to end tests to the configResetCmdF mmctl command #15982

### PR DESCRIPTION
**Pull request by Manimaran11**
#### Summary
This PR covers the addition of end to end tests to the configResetCmdF command

#### Ticket Link
Fixes mattermost/mattermost-server#15982

### CODE REVIEW

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @Manimaran11! :tada:

File: commands/config_e2e_test.go [hunk 1]
```diff
@@ -1,14 +1,56 @@
...
+	s.SetupTestHelper().InitBasic()
+
+	_, appErr := s.th.App.CreateUser(&model.User{Email: s.th.GenerateTestEmail(), Username: model.NewId(), Password: model.NewId()})
+	s.Require().Nil(appErr)
+
+	s.RunForSystemAdminAndLocal("System admin and local reset", func(c client.Client) {
+		printer.Clean()
+		s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.PrivacySettings.ShowEmailAddress = false })
+		defaultConfig := &model.Config{}
+		defaultConfig.SetDefaults()
```
Comment (ashishbhate) [inline commands/config_e2e_test.go:26]: The config created here (`defaultConfig`) isn't linked to the server configs. So when we compare the values in the `Require` statements below, we aren't comparing with the config values of the server. 

We'll need to use the `GetConfig` method of the client to get the config from the server and then compare the returned value to the value we set above.

This applies to the test below as well

File: commands/config_e2e_test.go [hunk 2]
```diff
@@ -1,14 +1,56 @@
...
 
 	"github.com/mattermost/mmctl/client"
 	"github.com/mattermost/mmctl/printer"
 )
 
+func (s *MmctlE2ETestSuite) TestConfigResetCmdE2E() {
+	s.SetupTestHelper().InitBasic()
+
+	_, appErr := s.th.App.CreateUser(&model.User{Email: s.th.GenerateTestEmail(), Username: model.NewId(), Password: model.NewId()})
+	s.Require().Nil(appErr)
```
Comment (ashishbhate) [inline commands/config_e2e_test.go:20]: Why are we creating a new user here? I dont think it is required.

**Review by ashishbhate — COMMENTED** [review]
Thanks for the effort @Manimaran11 ! A few comments inline

File: commands/config_e2e_test.go (hunk 1 above)
Comment (Manimaran11) [reply commands/config_e2e_test.go:26]: the config is compared with s.th.App.Config() in line 29.

File: commands/config_e2e_test.go (hunk 2 above)
Comment (Manimaran11) [reply commands/config_e2e_test.go:20]: will remove this

File: commands/config_e2e_test.go (hunk 1 above)
Comment (ashishbhate) [reply commands/config_e2e_test.go:26]: >the config is compared with s.th.App.Config() in line 29.

I think you mean line 35?


I'm still not sure why lines 25, 26 and 32 are required?

Comment (Manimaran11) [reply commands/config_e2e_test.go:26]: yeah..its 35,I was removing some lines in local, hence the confusion. Yeah, I think I can remove 25,26 and 32.

**Review by reflog — APPROVED** [review]

File: commands/config_e2e_test.go (hunk 1 above)
Comment (Manimaran11) [reply commands/config_e2e_test.go]: I have made the changes.

**Review by ashishbhate — APPROVED** [review]
LGTM! Thanks @Manimaran11

**Review by lindy65 — APPROVED** [review]
Thanks @Manimaran11 @mgdelacroix - looks good to me 👍

**Comment by hanzei** [comment]
Thank you for the contribution @Manimaran11 :tada:

**Pull request merged**